File: src/main/java/net/yogstation/yogbot/DatabaseManager.java
package net.yogstation.yogbot;

import com.mysql.cj.jdbc.MysqlConnectionPoolDataSource;
import net.yogstation.yogbot.config.DatabaseConfig;
import org.springframework.stereotype.Component;

import javax.naming.InitialContext;
import javax.sql.DataSource;
import java.sql.Connection;
import java.sql.SQLException;

@Component
public class DatabaseManager {
	private final DatabaseConfig config;
	private final MysqlConnectionPoolDataSource ds;

	public DatabaseManager(DatabaseConfig config) throws SQLException {
		this.config = config;
		ds = new MysqlConnectionPoolDataSource();
		ds.setServerName(config.hostname);
		ds.setPort(config.port);
		ds.setDatabaseName(config.database);
		ds.setUser(config.username);
		ds.setPassword(config.password);
	}

	public String prefix(String tableName) {
		return config.prefix + tableName;
	}

	public Connection getConnection() throws SQLException {
		return ds.getConnection();
	}
}

File: src/main/java/net/yogstation/yogbot/DatabaseManager.kt
package net.yogstation.yogbot

import net.yogstation.yogbot.config.DatabaseConfig
import com.mysql.cj.jdbc.MysqlConnectionPoolDataSource
import org.springframework.stereotype.Component
import java.sql.Connection
import kotlin.Throws
import java.sql.SQLException

@Component
class DatabaseManager(private val config: DatabaseConfig) {
	private val ds: MysqlConnectionPoolDataSource = MysqlConnectionPoolDataSource()

	init {
		ds.serverName = config.hostname
		ds.port = config.port
		ds.databaseName = config.database
		ds.user = config.username
		ds.password = config.password
	}

	fun prefix(tableName: String): String {
		return config.prefix + tableName
	}

	@get:Throws(SQLException::class)
	val connection: Connection
		get() = ds.connection
}
